aperturedb/ParallelQuery.py

```diff
@@ -0,0 +1,159 @@
+import math
+import time
+from threading import Thread
+
+from aperturedb import ProgressBar
+
+import numpy as np
+
+class ParallelQuery:
+
+    """
+        Parallel and Batch Querier for ApertureDB
+    """
+
+    def __init__(self, db, dry_run=False):
+
+        # db can me None in the case of Downloaders
+        # or other abstractions that want to use the parallel/batching
+        # structures without running queries
+        if db:
+            self.db = db.create_new_connection()
+
+        self.dry_run = dry_run
+
+        self.type = "query"
+
+        # Default Values
+        self.batchsize  = 1
+        self.numthreads = 1
+
+        self.total_queries = 0
+        self.times_arr = []
+        self.queries_time = 0
+        self.error_counter  = 0
+
+        self.loader_filename = "progress.log"
+
+    def  get_times(self):
+
+        return self.times_arr
+
+    def generate_batch(self, data):
+
+        """
+            Here we flatten the individual queries to run them as
```

Comment:
Are there other queries other than the Add stuff in loaders that would be using this ? All the loaders implement their own generate batch. Maybe an example in the docstring would explain the use case clearly.

Reply:
this is default behaviour of parallel queries: the generator is suppose to generate 1 query at the time, and we combine them to form a batch here. 
In the case of the loaders, we not only may combine queries, but also check some correctness dependent on the loader itself (for instace, the image loader checks that the query actually has the right parameters of AddImage and that there is a blob associated with it.)

Here is an example of a generator for a Find query:
https://github.com/aperture-data/athena-benchmark/pull/6/files#diff-8059b8cf2fdfd43fd87ce0d352f53c4dfa2deee54c4dfb058b1c0d142055c295R52

You are right, I should add more explanation to the docstring. 
I will add more comments

Reply:
OK. I see. I think I updated a top level comment about inheritance . Let me know if that makes sense.